apps/openmw/mwworld/worldimp.cpp

```diff
@@ -1024,9 +1025,17 @@ namespace MWWorld
 
             float activationDistance = getMaxActivationDistance() + telekinesisRangeBonus;
 
-            facedObject = getFacedObject(activationDistance);
+            facedObject = getFacedObject(activationDistance, distanceToObject, true);
+
+            // Not allowing telekinesis on actors, on doors that teleport to other cells, or on activators
+            // Original engine doesn't allow telekinesis on books or lights, either
+            if (!facedObject.isEmpty() && (facedObject.getClass().isActor()
+                || facedObject.getCellRef().getTeleport() || facedObject.getClass().getTypeName() == "struct ESM::Activator")
```

Comment:
Actually ```facedObject.getClass().getTypeName() == "struct ESM::Activator"``` would not work in the first place, the types are compiler-dependent. On Linux/GCC typeid names look like this (for an ESM::NPC for example):
<pre>
N3ESM3NPCE
</pre>